vcd/datasource_vcd_catalog_media_test.go

```diff
@@ -0,0 +1,157 @@
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { preRunChecks(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: catalogMediaDestroyed(testSuiteCatalogName, TestCatalogMediaDS),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:             configText,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdCatalogMediaExists("vcd_catalog_media."+TestAccVcdDataSourceMedia, &media),
+					resource.TestMatchOutput("owner_name", regexp.MustCompile(`^\w*$`)),
+					resource.TestMatchOutput("creation_date", regexp.MustCompile(`^(2019|2020)-`)),
+					resource.TestCheckOutput("status", "RESOLVED"),
+					resource.TestMatchOutput("storage_profile_name", regexp.MustCompile(`(.|\s)*\S(.|\s)*`)),
+					testCheckMediaNonStringOutputs(),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "vcd_catalog_media." + TestAccVcdDataSourceMedia + "-import",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdByCatalogMedia(TestAccVcdDataSourceMedia),
+				// These fields can't be retrieved from catalog media data
+				ImportStateVerifyIgnore: []string{"media_path", "upload_piece_size", "show_upload_progress"},
+			},
+		},
+	})
+}
+
+func catalogMediaDestroyed(catalog, mediaName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*VCDClient)
+		org, err := conn.GetOrgByName(testConfig.VCD.Org)
+		if err != nil {
+			return err
+		}
+		cat, err := org.GetCatalogByName(catalog, false)
+		if err != nil {
+			return err
+		}
+		_, err = cat.GetMediaByName(mediaName, false)
+		if err == nil {
+			return fmt.Errorf("catalog media %s not deleted", mediaName)
+		}
+		return nil
+	}
+}
+
+func importStateIdByCatalogMedia(objectName string) resource.ImportStateIdFunc {
+	return func(*terraform.State) (string, error) {
+		importId := testConfig.VCD.Org + "." + testSuiteCatalogName + "." + objectName
+		if testConfig.VCD.Org == "" || testSuiteCatalogName == "" || objectName == "" {
+			return "", fmt.Errorf("missing information to generate import path: %s", importId)
+		}
+		return importId, nil
+	}
+}
+
+const testAccCheckVcdCatalogMediaDS = `
+resource "vcd_catalog_media"  "{{.CatalogMediaName}}" {
+  org     = "{{.Org}}"
+  catalog = "{{.Catalog}}"
+
+  name                 = "{{.CatalogMediaName}}"
+  description          = "{{.Description}}"
+  media_path           = "{{.MediaPath}}"
+  upload_piece_size    = {{.UploadPieceSize}}
+  show_upload_progress = "{{.UploadProgress}}"
+
+  metadata = {
+    catalogMedia_metadata = "catalogMedia Metadata"
+    catalogMedia_metadata2 = "catalogMedia Metadata2"
+  }
+}
+
+data "vcd_catalog_media" "{{.NewCatalogMedia}}" {
+  org     = "{{.Org}}"
+  catalog = "{{.Catalog}}"
+  name    = "{{.CatalogMediaName}}"
+  depends_on = ["vcd_catalog_media.{{.CatalogMediaName}}"]
+}
```

Comment:
I was initially confused by this setting, because the `NewCatalogMedia` suggests something being created, but the `data` means that we are using an existing resource. Then I noticed that the `name` element is the same as the resource above.

Since we use a media item for govcd tests, meaning that our servers already have one, why not checking whether any item exists and use it, like we do for vApps and networks?

Reply:
But this create very strange dependency - you have to run govcd tests before terraform test. Also you have option ignore configuring media in govcd and test will be ignored.

Reply:
What I mean is that we have already a media item created and pre-installed for the purpose of using it with govcd tests. The check I suggest should look for existing media and use it for data source testing.
With our pre-configured servers, the media item is always there.

Reply:
In current configuration we don't have media reference.